src/main/java/lotto/domain/LottoNumbers.java

```diff
@@ -0,0 +1,50 @@
+package lotto.domain;
+
+import lotto.exception.IllegalLottoNumberCountException;
+import lotto.util.LottoNumber;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class LottoNumbers {
+
+    private final List<Integer> lottoNumbers;
+
+    public LottoNumbers(List<Integer> lottoNumbers) {
+        if (!isValid(lottoNumbers)) {
+            throw new IllegalLottoNumberCountException(lottoNumbers.size());
+        }
+        this.lottoNumbers = lottoNumbers;
+    }
+
+    public static LottoNumbers of(String separateNumber) {
+        SeparateNumber number = SeparateNumber.of(separateNumber);
+        return new LottoNumbers(number.initLottoNumbers());
+    }
+
+    public List<Integer> getLottoNumbers() {
+        return Collections.unmodifiableList(lottoNumbers);
+    }
+
+    public int match(LottoNumbers lottoTickets) {
+        List<Integer> buyTicket = lottoTickets.getLottoNumbers();
+        int sumValue = 1;
+        int startIndex = 0;
+        return buyTicket.stream()
+                .filter(this.lottoNumbers::contains)
+                .map(e -> sumValue).reduce(startIndex, Integer::sum);
```

Comment:
```suggestion
                .map(e -> sumValue)
                .reduce(startIndex, Integer::sum);
```
개행이 필요할 것 같아요!

Reply:
```suggestion
                .count()
```
다시보니 `count`만 사용해도 될 것 같네요

Reply:
count 를 하면 리턴값이 long 이여서 일부러 저렇게 해봤는데 한번 다시 고민해보겠습니다.